documentation/Outcomes-Display.md

```diff
@@ -0,0 +1,427 @@
+
+The following fields are also configurable in "outcomes_display" layouts:
+
+- `title`: the name of the "outcome" that shows in the "Outcomes" menu and as the Header of the table (e.g. "Users"). Default is just the tableName converted to plural Title Case, e.g. `user` => "Users"
+- `permissionNames`: an array of permissions names that the request requires (via JWT) in order to see this Layout. Default is an empty array, and both that and `null` allow full access (i.e. no restrictions)
+- `table_view_include_columns`: an array of fields to return for the "table" endpoint. Should be field names written in camelCase. All values should match with either the name of a field in the table being queried, or the name of a "custom" column defined in `outcome_display_column_definition` (see below).  
+Some things to be aware of:
+  - `null` or empty array means return ALL fields
+  - if you've named a "custom" field in here and yet still want to return all other fields, you can specify this by including the special value "..." as an item, which has the same effect as the "...rest" parameter in javascript.
+- `table_view_exclude_columns`: an array of columns to exclude. Basically, it takes all the columns collected from the "include" list, and removes any named here.
+- `detail_view_include_columns` / `detail_view_exclude_columns`: exactly the same as the "table_view" fields, except applies to the fields that show up in the Details view
+- `detail_view_header_column`: name of the field/column whose value should be used as the Header display in Details view. If not specified, the table name will just be used.
+- `show_linked_applications`: if `true`, the `/item` endpoint will also return an array of linked applications connected to this particular item. These are taken from the outcome "join" tables, whose records are created on successful application approvals.
+- `priority`: when multiple layouts match the request (i.e. user has permissions for more than one Layout), the returned columns will be a union of the columns specified in both layouts. However,there will still only be one "Title" and "Header column" returned and each layout may have a different one, so the layout with the highest priority will be used. In general you'd want to give the more "restrcited" permission the higher priority, but most of the time you won't need to consider this as the Title and Header would often be the same. Default value: `1`.
+
+Okay, so if you're just wanting to display fields directly taken from the outcome table in question, and the formatting requirements are all "simple" types (text, number, boolean, Date) the `outcomes_display` table is all you need. However, if you want to return columns with more complex data (such as a list of ingredients, or a query to another table) or require non-default formatting, you'll need to define these in the `outcome_display_column_definition` table.
+
+### `outcome_display_column_definition` table
```

Comment:
I'm a little lost in the structure here (maybe a diagram would help). Let me check this is accurate:
- `outcome_table` is defining the `displayDefinitions` which can be defining a title per column  for each column displayed in the list of entities to be displayed for this outcome.
- `outcomes_display ` specify how the column header  to show in a column when displaying one entity using `detail_view_header_column`
  just be using the field name (as my query before - which can be displayed as  Camel case with spaces...).
- `outcome_display_ column_definition` can also define more fine grained header to show in a column when displaying one entity.

Reply:
I'm not entirely clear what you're referring to. Are you perhaps getting mixed up between the two definition tables, and the structure of the returned data? I'm not sure which one you're asking about here.

Reply:
Can you explain which part you're having trouble with and I'll try and clarify it, but I honestly can't think what a diagram here would even look like -- it's (almost) a single table of fields, and I've tried hard to start at the simplest possible cases and work up to more complexity with examples.

Is it the configuration, or the output data structure you're confused about?